lib/msal-angular/src/msal.interceptor.ts

```diff
@@ -122,19 +140,62 @@ export class MsalInterceptor implements HttpInterceptor {
             } else {
                 // Matching endpoint with both protected resource and relative url of protected resource
-                return StringUtils.matchPattern(normalizedKey, normalizedEndpoint) || StringUtils.matchPattern(relativeNormalizedKey, normalizedEndpoint);
+                return StringUtils.matchPattern(normalizedKey, endpoint) || StringUtils.matchPattern(relativeNormalizedKey, endpoint);
             }
         });
+    }
 
-        // Process all protected resources and send the first matched resource
-        if (keyMatchesEndpointArray.length > 0) {
-            const keyForEndpoint = keyMatchesEndpointArray[0];
-            if (keyForEndpoint) {
-                return this.msalInterceptorConfig.protectedResourceMap.get(keyForEndpoint);
+    /**
+     * Finds scopes from first matching endpoint with HTTP method that matches request
+     * @param protectedResourceMap Protected resource map
+     * @param endpointArray Array of resources that match request endpoint
+     * @param httpMethod Http method of the request
+     * @returns 
+     */
+    static matchScopesToEndpoint(protectedResourceMap: Map<string, Array<string|ProtectedResourceScopes> | null>, endpointArray: string[], httpMethod: string): Array<string>|null {
+        const allMatchedScopes = [];
+
+        // Check each matched endpoint for matching HttpMethod and scopes
+        endpointArray.forEach(matchedEndpoint => {
+            const scopesForEndpoint = [];
+            const methodAndScopesArray = protectedResourceMap.get(matchedEndpoint);
+
+            // Return if resource is unprotected
+            if (methodAndScopesArray === null) {
+                allMatchedScopes.push(null);
+                return;
+            }
+
+            methodAndScopesArray.forEach(entry => {
+                // Entry is either array of scopes or ProtectedResourceScopes object
+                if (typeof entry === "string") {
+                    scopesForEndpoint.push(entry);
+                } else {
+                    // Ensure methods being compared are normalized
+                    const normalizedRequestMethod = httpMethod.toLowerCase();
+                    const normalizedResourceMethod = entry.httpMethod.toLowerCase();
+
+                    // Method in protectedResourceMap matches request http method
+                    if (normalizedResourceMethod === normalizedRequestMethod) {
+                        entry.scopes.forEach(scope => {
+                            scopesForEndpoint.push(scope);
+                        });
+                    }
+                }
+            });
+
+            // Only add to all scopes if scopes for endpoint and method is found
+            if (scopesForEndpoint.length > 0) {
+                allMatchedScopes.push(scopesForEndpoint);
             }
+        });
+
+        if (allMatchedScopes.length > 0) {
+            // Returns scopes for first matching endpoint
```

Comment:
I think it may be worth adding a warning log message here is more than 1 matched scopes are found.

Reply:
Sure. Added.